internal/ir/builder.go

```diff
@@ -233,12 +233,63 @@ func (b *builder) assign(fn *Function, lhs, rhs ast.Expr, syntax *ast.AssignStmt
 
 			return
 		}
+
+		if rhs, isIdent := rhs.(*ast.Ident); isIdent {
+			b.identAssign(fn, lhs, rhs, syntax)
+
+			return
+		}
+
 		fn.addNamedLocal(lhs.Name, b.expr(fn, rhs, false /*expand*/), syntax)
 	default:
 		unsupportedNode(lhs)
 	}
 }
 
+// identAssign responsible for handling rhs values that are identifiers. The values of these identifiers can be local
+// variables of a function, global variables or parameters of the function. Values within the function's context are
+// always prioritized over external values. Below is an example of code and IR for each of the scenarios listed.
+//
+// source code:
+//
+// const a = "x"
+//
+// 	function f1(b) {
+//    	const c = "x"
+//    	const d = c
+//    	const e = b
+//    	const f = a
+//	}
+//
+// IR representation:
+//
+//	c = "y"
+//	d = "y"
+//	e = nil
+//	f = "x"
+//
+func (b *builder) identAssign(fn *Function, lhs, rhs *ast.Ident, syntax *ast.AssignStmt) {
+	if local, ok := fn.Locals[rhs.Name]; ok {
```

Comment:
Use `fn.lookup` instead to get a function local variable.

Reply:
Done!